pkg/uio/reader_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uio
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+func readAndCheck(t *testing.T, want, tmpfileP string) {
+	t.Helper()
+	r := strings.NewReader(want)
+	if err := ReadIntoFile(r, tmpfileP); err != nil {
+		t.Errorf("ReadIntoFile(%v, %s) = %v, want no error", r, tmpfileP, err)
+	}
+
+	got, err := ioutil.ReadFile(tmpfileP)
+	if err != nil {
+		t.Fatalf("ioutil.ReadFile(%s) = %v, want no error", tmpfileP, err)
+	}
+	if want != string(got) {
+		t.Errorf("got: %v, want %s", string(got), want)
+	}
+}
+
+func TestReadIntoFile(t *testing.T) {
+	want := "I am the wanted"
+
+	// Write to a file already exist.
+	tmpfile, err := ioutil.TempFile("", "uio-out")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmpfileP := tmpfile.Name()
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	readAndCheck(t, want, tmpfileP)
+
+	// Write to a file that does not exist.
+	os.Remove(tmpfileP)
```

Comment:
I really hate to picky here, but ...  you are creating a file, removing it, then creating it again, assuming nobody else will sneak in and create it in the meantime. 

this is a race condition. It will never happen. Really. We all know it will never happen. but ... 

You'd be safer using TempDir and then creating files in that directory. 

btw, blame linux; /tmp should not be shared between users,or even sessions this has been known for at least 40 years, linux shoulda fixed it but ... did not. so we get stuck making TempDir ... geez.

Reply:
sorry I missed this before, I was worried I had approved too soon once I noticed Chris' comment, then notice this.

Reply:
That is a good point. Read a bit more online, a brief wikipedia page talks about the same suggestion as well https://en.wikipedia.org/wiki/Temporary_folder. 

Before making the change, for curiosity as always :), I looked into the source of `os.TempDir`, and `io.TempFile`, and found that the behavior of accessing / creating tempdir is the same

`ioutil.TempFile("", pattern)`  calls `os.TempDir`

https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/os/tempfile.go;l=32;drc=refs%2Ftags%2Fgo1.17

https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/os/file.go;l=389

Regardless, this is sth new learnt :)